python/dgl/convert.py

```diff
@@ -785,89 +779,326 @@ def to_homo(G):
-        Integer ID type. Must be int32 or int64. Default: int64.
+        The name of the edge attribute that stores the edge IDs. If given, DGL will assign edge
+        IDs accordingly when creating the graph, so the attribute must be valid IDs, i.e.
+        consecutive integers starting from zero. By default, the edge IDs of the returned graph
+        can be arbitrary. It must be None if :attr:`nx_graph` is undirected.
+    idtype : int32 or int64, optional
+        The data type for storing the structure-related graph information such as node and
+        edge IDs. It should be a framework-specific data type object (e.g., ``torch.int32``).
+        By default, DGL uses int64.
+    device : device context, optional
+        The device of the resulting graph. It should be a framework-specific device object
+        (e.g., ``torch.device``). By default, DGL stores the graph on CPU.
 
     Returns
     -------
-    g : DGLGraph
+    DGLGraph
+        The created graph.
+
+    Notes
+    -----
+    DGL internally maintains multiple copies of the graph structure in different sparse
+    formats and chooses the most efficient one depending on the computation invoked.
+    If memory usage becomes an issue in the case of large graphs, use
+    :func:`dgl.DGLGraph.formats` to restrict the allowed formats.
+
+    Examples
+    --------
+
+    The following example uses PyTorch backend.
+
+    >>> import dgl
+    >>> import networkx as nx
+    >>> import numpy as np
+    >>> import torch
+
+    Create a 2-edge NetworkX graph.
+
+    >>> nx_g = nx.DiGraph()
+    >>> # Add 3 nodes and two features for them
+    >>> nx_g.add_nodes_from([0, 1, 2], feat1=np.zeros((3, 1)), feat2=np.ones((3, 1)))
+    >>> # Add 2 edges (1, 2) and (2, 1) with two features, one being edge IDs
+    >>> nx_g.add_edge(1, 2, weight=np.ones((1, 1)), eid=np.array([1]))
+    >>> nx_g.add_edge(2, 1, weight=np.ones((1, 1)), eid=np.array([0]))
+
+    Convert it into a DGLGraph with structure only.
+
+    >>> g = dgl.from_networkx(nx_g)
+
+    Retrieve the node/edge features of the graph.
+
+    >>> g = dgl.from_networkx(nx_g, node_attrs=['feat1', 'feat2'], edge_attrs=['weight'])
+
+    Use a pre-specified ordering of the edges.
+
+    >>> g.edges()
+    (tensor([1, 2]), tensor([2, 1]))
+    >>> g = dgl.from_networkx(nx_g, edge_id_attr_name='eid')
+    (tensor([2, 1]), tensor([1, 2]))
+
+    Create a graph on the first GPU with data type int32.
+
+    >>> g = dgl.from_networkx(nx_g, idtype=torch.int32, device='cuda:0')
+
+    See Also
+    --------
+    graph
+    from_scipy
     """
-    # Relabel nodes using consecutive integers
+    # Sanity check
+    utils.check_type(nx_graph, nx.Graph, 'nx_graph', skip_none=False)
+    utils.check_all_same_type(node_attrs, str, 'node_attrs', skip_none=True)
+    utils.check_all_same_type(edge_attrs, str, 'edge_attrs', skip_none=True)
+    utils.check_type(edge_id_attr_name, str, 'edge_id_attr_name', skip_none=True)
```

Comment:
One more thing. Checking/conditioning on argument types is not a very pythonic thing to do. In Python, it matters more if an object *behaves* like some certain type rather than being actually of that type (a.k.a. duck type system).